src/packagedcode/phpcomposer.py

```diff
@@ -336,3 +354,56 @@ def parse_person(persons):
                 url and url.strip('() '))
     else:
         raise ValueError('Incorrect PHP composer persons: %(persons)r' % locals())
+
+
+def build_package_from_lock(package_data):
+    """
+    Yield a composer Package object from a package data mapping that originated
+    from a composer.lock file
```

Comment:
Is the logic to return one package for each entry of the lock file?

Reply:
Yes. I report every package listed there.